src/components/ble/SimpleWeatherService.h

```diff
@@ -61,13 +61,42 @@ namespace Pinetime {
         Unknown = 255
       };
 
+      class Temperature {
+      public:
+        explicit Temperature(int16_t raw = 0) : raw {raw} {
```

Comment:
I'm not a big fan of having the default value be 0 here, but otherwise the compiler complains about there being no default value for the array of `Day`s in the `Forecast`.

Reply:
I think it would be conceptually better (maybe not memory-wise) to make the days array of type:
`std::array<std::optional<Day>, nbOfDays>`. Then you would not need to construct `Day`s without values.

Reply:
Would defining Temperature in Kelvin internally and converting to DisplayTemperature in °C or °F depending on the user setting for displaying be helpful?
Sorry, I'm not into object based programming that much...

Reply:
I don't think that is the problem here. The internal representation is never shown directly and can therefore be whatever. The problem is, that right now we need a temperature value in a place where no value should be.

Reply:
I think `std::optional` does make sense, I'm going to see how implementing it like that looks.

Reply:
I think this works quite well, what do you think?